### PR TITLE
DHFPROD-7560: Add options to clear user data

### PR DESCRIPTION
**Pull request by aniamy**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests
- [x] Added to Release Wiki

### CODE REVIEW

File: marklogic-data-hub-central/ui/src/components/header/system-info.test.tsx [hunk 1]
```diff
@@ -58,19 +68,20 @@ describe("Update data load settings component", () => {
     expect(getByText("Delete all user data in the STAGING, FINAL, and JOBS databases. Project files and artifacts remain.")).toBeInTheDocument();
     expect(getByTestId("downloadProjectFiles")).toBeDisabled();
     expect(getByTestId("downloadHubCentralFiles")).toBeDisabled();
-    // TODO DHFPROD-7711 skipping failing checks to enable component replacement
-    // expect(getByText("Clear")).toBeDisabled();
+    expect(getByTestId("clearUserData")).toBeDisabled();
```
Comment (briantang) [inline marklogic-data-hub-central/ui/src/components/header/system-info.test.tsx:71]: Nice, thanks for this.

File: marklogic-data-hub-central/ui/src/components/header/system-info.test.tsx [hunk 2]
```diff
@@ -58,19 +68,20 @@ describe("Update data load settings component", () => {
...
     const {getByText, getByTestId} = render(<Router><AuthoritiesContext.Provider value={authorityService}>
       <SystemInfo {...data.environment}
         systemInfoVisible={true}
         setSystemInfoVisible={jest.fn()}
       />
     </AuthoritiesContext.Provider></Router>);
+
     expect(getByTestId("downloadProjectFiles")).toBeEnabled();
     expect(getByTestId("downloadHubCentralFiles")).toBeEnabled();
     // expect(getByText("Clear")).toBeDisabled();
```
Comment (briantang) [inline marklogic-data-hub-central/ui/src/components/header/system-info.test.tsx:87]: Can we do something similar and uncomment this to use the test id?

File: marklogic-data-hub-central/ui/src/components/header/system-info.test.tsx [hunk 3]
```diff
@@ -106,13 +118,64 @@ describe("Update data load settings component", () => {
...
+      fireEvent.click(getByText("Entity"));
+      const entitiesInput = getByPlaceholderText("Search entities");
+      expect(entitiesInput).toBeInTheDocument();
+      expect(getByText("Clear")).not.toBeEnabled();
+
+      userEvent.type(entitiesInput, "cust");
+      expect(getByLabelText("Customer")).toBeInTheDocument();
+      fireEvent.click(getByLabelText("Customer"));
+      expect(getByText("Clear")).toBeEnabled();
+
```
Comment (briantang) [inline marklogic-data-hub-central/ui/src/components/header/system-info.test.tsx:175]: The RTL test looks good and so close to perfect! Is it possible to add one last check of clicking the "Clear" button and verifying the endpoint is called with the collection and database in the payload? Something like:
`    expect(axiosMock.post).toBeCalledWith("/api/environment/clearUserData", payload);
`
where 
`payload = { targetCollection: "Customer", targetDatabase: "STAGING"}`

**Review by briantang — COMMENTED** [review]
I see the text underneath delete subset is greyed out when "Clear All" is selected, but the text underneath "Clear All" is not greyed out when "Delete Subset" is selected. Can we grey out that case so it is consistent?

File: marklogic-data-hub-central/ui/src/components/header/system-info.test.tsx (hunk 3 above)
Comment (aniamy) [reply marklogic-data-hub-central/ui/src/components/header/system-info.test.tsx:174]: Added. I had some issue when checking the modal text. But left a comment to include it once the updated text story is worked on.

**Review by ryanjdew — APPROVED** [review]

**Review by briantang — APPROVED** [review]
Looks good now!

**Pull request merged**